crates/sui-framework/packages/deepbook/sources/clob_v2.move

```diff
@@ -166,6 +166,22 @@ module deepbook::clob_v2 {
         maker_rebates: u64
     }
 
+    /// Emitted only when a taker order is filled.
+    struct TakerBalanceChange<phantom BaseAsset, phantom QuoteAsset> has copy, store, drop {
```

Comment:
It might just be the naming, but this PR does not capture all taker order filled events. Taker orders by definition also include limit orders that match with an existing orders, but I don't see `TakerBalanceChange` emitted for these cases(i.e., `match_ask` and `match_bid`).

Reply:
`match_ask` and `match_bid` are actually called under the place_market_order function so we actually cover both cases in this change.

Reply:
Updated code to emit proper changes on limit order fills